app.js

```diff
@@ -29,6 +29,10 @@ app.get("/newmovies", async (req, res) => {
 
 app.get("/newmovies/:movieId", async (req, res) => {
   const movie = await loadMovie(req.params.movieId);
+  if (!movie) {
+    res.status(404).redirect("/404");
+    return;
+  }

```

Comment:
This is confusing, because first you set status to 404, but then you use `redirect()` which will actually respond with a HTTP 302. Have a look at the response in your dev tools and you will see what I mean.

<img width="739" alt="image" src="https://github.com/oskber/kino-a5/assets/550212/f7ae49c2-b5e1-4e54-a84b-7c21f1453087">

Instead of redirecting, you should just respond with 404 and the content for the 404 page here. Something like:

```js
if (!movie) {
  res.status(404).render("404");
  return;
}
```